google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,367 @@
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param managerId the root customer ID from which to begin the search.
+   * @param loginCustomerId the loginCustomerId used to create the GoogleAdsClient.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, Long managerId, Long loginCustomerId) {
+    List<Long> seedCustomerIds = new ArrayList<>();
+    if (managerId != null) {
+      seedCustomerIds.add(managerId);
+    } else {
+      seedCustomerIds = getAccessibleCustomers(googleAdsClient);
+    }
+
+    Map<CustomerClient, Multimap<Long, CustomerClient>> allHierarchies = new HashMap<>();
+    List<Long> accountsWithNoInfo = new ArrayList<>();
+    // Constructs a map of account hierarchies.
+    for (long seedCustomerId : seedCustomerIds) {
+      Map<CustomerClient, Multimap<Long, CustomerClient>> customerClientToHierarchy =
+          createCustomerClientToHierarchy(loginCustomerId, seedCustomerId);
+
+      if (customerClientToHierarchy != null) {
+        allHierarchies.putAll(customerClientToHierarchy);
+      } else {
+        accountsWithNoInfo.add(seedCustomerId);
+      }
+    }
+
+    // Prints the IDs of any accounts that did not produce hierarchy information.
+    if (!accountsWithNoInfo.isEmpty()) {
+      System.out.println(
+          "Unable to retrieve information for the following accounts which are likely either test "
+              + "accounts with setup issues. Please check the logs for additional details.");
+      for (long accountId : accountsWithNoInfo) {
+        System.out.println(accountId);
+      }
+      System.out.println();
+    }
+
+    int depth = 0;
+    // Prints the hierarchy information for all accounts for which there is hierarchy information
+    // available.
+    for (CustomerClient rootCustomerClient : allHierarchies.keySet()) {
+      System.out.printf(
+          "The hierarchy of customer ID %d is printed below.%n",
+          rootCustomerClient.getId().getValue());
+      printAccountHierarchy(rootCustomerClient, allHierarchies.get(rootCustomerClient), depth);
+      System.out.println();
+    }
+  }
+
+  /**
+   * Creates a new GoogleAdsClient instance with the specified loginCustomerId.
+   *
+   * @param loginCustomerId the loginCustomerId used to create the GoogleAdsClient.
+   * @return a GoogleAdsClient instance. Returns null if the GoogleAdsClient cannot be created.
+   */
+  private GoogleAdsClient createGoogleAdsClient(long loginCustomerId) {
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient =
+          GoogleAdsClient.newBuilder()
+              .fromPropertiesFile()
+              .setLoginCustomerId(loginCustomerId)
+              .build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file with loginCustomerId %d. " +
+            "Exception: %s%n", loginCustomerId, fnfe);
+      return null;
```

Comment:
Having a second thought, I don't think this makes sense here.
If the property file cannot be found, it should fail at the step of creating default GAC above.
Even if the IO fails somehow, we should probably fail fast and tell the example users that something is gone wrong, so they can go fix it.

In other words, what's the benefit of printing "your prop file cannot be found" a hundred times?
They still need to fix the prop file. They probably don't care that we print that "your prop file cannot be found" for all the customers that their Google account has access to.

Reply:
See revised solution.